enterprise/frontend/src/metabase-enterprise/whitelabel/components/FontWidget/FontWidget.tsx

```diff
@@ -1,55 +1,58 @@
-import { useCallback, useMemo } from "react";
 import { t } from "ttag";
 
-import MetabaseSettings from "metabase/lib/settings";
-import { Select } from "metabase/ui";
-
-import type { FontSetting, FontSettingKeys, FontSettingValues } from "./types";
-
-export interface FontWidgetProps {
-  setting: FontSetting;
-  settingValues: FontSettingValues;
-  availableFonts?: string[];
-  onChange: (value: string) => void;
-  onChangeSetting: (key: FontSettingKeys, value: unknown) => void;
-}
-
-const CUSTOM = "custom";
-
-const FontWidget = ({
-  setting,
-  settingValues,
-  availableFonts = MetabaseSettings.get("available-fonts") || [],
-  onChange,
-  onChangeSetting,
-}: FontWidgetProps): JSX.Element => {
-  const value = !settingValues["application-font-files"]
-    ? (setting.value ?? setting.default)
-    : CUSTOM;
-
-  const options = useMemo(
-    () => [
-      ...availableFonts.map((font) => ({ label: font, value: font })),
-      { label: t`Custom…`, value: CUSTOM },
-    ],
-    [availableFonts],
+import { SettingHeader } from "metabase/admin/settings/components/SettingHeader";
+import { BasicAdminSettingInput } from "metabase/admin/settings/components/widgets/AdminSettingInput";
+import { useAdminSetting } from "metabase/api/utils";
+import { Box } from "metabase/ui";
+
+import { FontFilesWidget } from "./FontFilesWidget";
+import { useGetFontOptions } from "./utils";
+
+const defaultFont = "Lato";
+
+export const FontWidget = () => {
+  const fontOptions = useGetFontOptions();
+  const {
+    value: font,
+    updateSetting,
+    description,
+  } = useAdminSetting("application-font");
+  const { value: fontFiles } = useAdminSetting("application-font-files");
+
+  const fontValue = fontFiles ? "custom" : font;
+
+  const handleChange = async (newValue: string) => {
+    if (newValue === fontValue) {
+      return;
+    }
+
+    await updateSetting({
+      key: "application-font",
+      value: newValue === "custom" ? defaultFont : newValue,
+    });
+
+    await updateSetting({
+      key: "application-font-files",
+      value: newValue === "custom" ? [] : null,
+      toast: false,
+    });
+  };
+
+  return (
+    <Box>
+      <SettingHeader
+        id="application-font"
+        title={t`Font`}
+        description={description}
+      />
```

Comment:
I think we need to handle the SetByEnvVar stuff here (which gets automatically handled in `AdminSettingInput`? We have [MB_APPLICATION_FONT](https://www.metabase.com/docs/latest/configuring-metabase/environment-variables#mb_application_font)